c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -198,12 +200,14 @@ struct Block {
       int device,
       cudaStream_t stream,
       size_t size,
+      size_t requested_size,
       BlockPool* pool,
       void* ptr)
       : device(device),
         stream(stream),
         stream_uses(),
         size(size),
+        requested_size(requested_size),
```

Comment:
We can initialize this to 0 and set it to to the requested size when the block becomes allocated.